File: Backend/VestTour.Domain/Entities/Product.cs
using System;
using System.Collections.Generic;

namespace VestTour.Domain.Entities;

public partial class Product
{
    public int ProductId { get; set; }

    public string? ProductCode { get; set; }

    public int? MeasurementId { get; set; }

    public int? CategoryId { get; set; }

    public int? FabricId { get; set; }

    public int? LiningId { get; set; }

    public bool? IsCustom { get; set; }

    public string? ImgUrl { get; set; }

    public decimal? Price { get; set; }

    public string? Size { get; set; }

    public string? Status { get; set; }

    public virtual Category? Category { get; set; }

    public virtual Fabric? Fabric { get; set; }

    public virtual ICollection<Feedback> Feedbacks { get; set; } = new List<Feedback>();

    public virtual Lining? Lining { get; set; }

    public virtual Measurement? Measurement { get; set; }

    public virtual ICollection<OrderDetail> OrderDetails { get; set; } = new List<OrderDetail>();

    public virtual ICollection<StyleOption> StyleOptions { get; set; } = new List<StyleOption>();
}

File: Mobile/mobileapp/lib/models/product.dart
class Product {
  int? productID;
  String? productCode;
  int? measurementID;
  int? categoryID;
  int? fabricID;
  int? liningID;
  String? size;
  bool? isCustom;
  String? imgURL;
  double? price;

  Product(
      {this.productID,
        this.productCode,
        this.measurementID,
        this.categoryID,
        this.fabricID,
        this.liningID,
        this.size,
        this.isCustom,
        this.imgURL,
        this.price});

  Product.fromJson(Map<String, dynamic> json) {
    productID = json['productID'];
    productCode = json['productCode'];
    measurementID = json['measurementID'];
    categoryID = json['categoryID'];
    fabricID = json['fabricID'];
    liningID = json['liningID'];
    size = json['size'];
    isCustom = json['isCustom'];
    imgURL = json['imgURL'];
    price = json['price'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['productID'] = this.productID;
    data['productCode'] = this.productCode;
    data['measurementID'] = this.measurementID;
    data['categoryID'] = this.categoryID;
    data['fabricID'] = this.fabricID;
    data['liningID'] = this.liningID;
    data['size'] = this.size;
    data['isCustom'] = this.isCustom;
    data['imgURL'] = this.imgURL;
    data['price'] = this.price;
    return data;
  }
}
